src/test/mochitest/examples/babel/source-maps-semantics.md

```diff
@@ -0,0 +1,49 @@
+# Source Map Semantics
+
+Source Maps are a specification for helping users how their original maps onto a single file.
+While the mapping specifies to point-to-point relationships, 
+it is up to source map generators to decides the appropriate relationships in the code.
+
+This document attempts to outline some of the existing mappings we have observed 
+as well as articulate some opinions on the appropriate mappings.
+
+### Calls
+
+The call maps the identifier and includes the opening original paren.
+
+![](https://shipusercontent.com/554e9a9d7b74a52465156d80391625f8/Screen%20Shot%202018-03-28%20at%2010.13.15%20PM.png)
+
+The object property of a member expression call maps the identifier, period, and generated parens.
+
+![](https://shipusercontent.com/17b1f5147336473f9d5070d5d7b82094/Screen%20Shot%202018-03-28%20at%2010.13.11%20PM.png)
+
+The arguments of a call attempt to map one to one. The last argument of a call will include the closing original paren.
+
+![](https://shipusercontent.com/33ff9ea55c3cd35d55b70e64cf9326e9/Screen%20Shot%202018-03-28%20at%2010.13.01%20PM.png)
+
+Complex arguments are also mapped one to one. In this case, the argument itself has a call that is included in the mapping.
+
+![](https://shipusercontent.com/20f575759f31c01d13243e83a253c9ba/Screen%20Shot%202018-03-28%20at%2010.16.20%20PM.png)
+
+
+### Dependencies
+
+When Babel is involved, most dependencies are mapped line by line, but namespaces are broken up.
+
+example: ![imported-bindings]
+
+![](https://shipusercontent.com/c0939aecadad12902646bd687b05840f/Screen%20Shot%202018-03-28%20at%2010.00.30%20PM.png)
+
+![](https://shipusercontent.com/fadbe965f836d304e84989f89529276e/Screen%20Shot%202018-03-28%20at%2010.07.57%20PM.png)
+![](https://shipusercontent.com/cc5ff07e169e2583e9e4e8859b912810/Screen%20Shot%202018-03-28%20at%2010.07.52%20PM.png)
+
+When Webpack is left to its own devices then dependencies are not mapped at all. In these cases, the original dependencies all map to the generated location `0,0`, which is equivelant to null.
+
+- example 1: ![webpack-es-modules]  
+- example 2: ![webpack-modules]
```

Comment:
@loganfsmyth what's the difference between es-modules and modules here? I always forget :)

we might want to put a readme in some of the examples.

Reply:
Fair. The `es` one runs with no Babel plugins, and the non-es one runs with preset-env, so the `es` one still has ES20XX features in it.